### PR TITLE
fp: Plug in MPFR floating-point literal back end.

### PR DESCRIPTION
**Pull request by aniemetz**
This does not yet enable MPFR as backend for floating-point literals yet. It also does
not yet enable MPFR in CI builds, this will come as a separate step where we will
configure CMake to build MPFR locally if not found in the system.

This further introduces an ENABLE_SLOW_TESTS switch in the unit tests that can
be flipped when testing locally.

### CODE REVIEW

File: test/unit/util/floatingpoint_black.cpp [hunk 1]
```diff
@@ -781,8 +791,10 @@ TEST_F(TestUtilBlackFloatingPoint, convertToBV)
     auto mpfr = fpMPFR(fmt, bv);
     auto sym = fpSymFPU(fmt, bv);
 
-    ASSERT_EQ(mpfr.convertToSBVTotal(width, rm, undef),
-              sym.convertToSBVTotal(width, rm, undef));
+    // SymFPU to_sbv has an issue for a corner case, see #12734 that may
+    // get triggered in this test, thus it is temporarily disabled.
+    // ASSERT_EQ(mpfr.convertToSBVTotal(width, rm, undef),
+    //           sym.convertToSBVTotal(width, rm, undef));
```
[the remark this answers is not in the document]
Comment (aniemetz) [reply test/unit/util/floatingpoint_black.cpp:805]: Wontfix, does not comply with our coding guidelines.

File: test/unit/util/floatingpoint_black.cpp [hunk 2]
```diff
@@ -32,10 +32,11 @@ namespace test {
 class TestUtilBlackFloatingPoint : public TestInternal
 {
  protected:
+  static constexpr bool ENABLE_SLOW_TESTS = false;
```
[the remark this answers is not in the document]
Comment (aniemetz) [reply test/unit/util/floatingpoint_black.cpp]: I will disable the lines for now, and will enable them in the follow-up PR #12736.

File: src/util/floatingpoint_literal.cpp [hunk 3]
```diff
@@ -13,15 +13,24 @@
...
+
 /* -------------------------------------------------------------------------- */
 
 namespace cvc5::internal {
 
+#ifdef CVC5_USE_MPFR
+using FPLit = FloatingPointLiteralMPFR;
+#else
 using FPLit = FloatingPointLiteralSymFPU;
+#endif
```
[the remark this answers is not in the document]
Comment (aniemetz) [reply src/util/floatingpoint_literal.cpp]: I will disable the lines for now, and will enable them in the follow-up PR https://github.com/cvc5/cvc5/pull/12736.

**Review by mpreiner — APPROVED** [review]
